twisted/conch/ssh/keys.py

```diff
@@ -1194,12 +1380,33 @@ def verify(self, signature, data):
                 k = k.public_key()
             verifier = k.verifier(
                 signature, hashes.SHA1())
+        elif name == 'EC':
+            concatenatedSignature = common.getNS(signature)[0]
+            rstr, sstr, rest = common.getNS(concatenatedSignature, 2)
+            r = int_from_bytes(rstr, 'big')
+            s = int_from_bytes(sstr, 'big')
+            signature = encode_dss_signature(r, s)
+
+            k = self._keyObject
+            if not self.isPublic():
+                k = k.public_key()
+
+            #Which hash to use depends on the key size.
+            if self._keyObject.curve.key_size <= 256:
```

Comment:
This code is repeated from above.  I think we need a `_curve2hash` utility function or something.  Although I'm almost relieved that this is the case: when you have a big if/elif tree that all sets the same variable, it's nicer to encapsulate it as a function, because `h = foo()` only sets `h` once, whereas if you get a big tree you have to ensure that you didn't misspell `h` in any of its branches when you read it.